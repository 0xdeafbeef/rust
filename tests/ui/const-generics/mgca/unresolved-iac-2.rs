#![feature(min_generic_const_args)]
#![feature(inherent_associated_types)]
#![expect(incomplete_features)]

struct Foo<'a> {
    x: &'a (),
}

impl<'a> Foo<'a> {
    fn foo(_: [u8; Foo::X]) {}
    //~^ ERROR: associated constant `X` not found for `Foo<'_>` in the current scope
}

fn main() {}
