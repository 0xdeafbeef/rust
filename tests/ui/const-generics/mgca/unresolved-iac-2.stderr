error[E0220]: associated constant `X` not found for `Foo<'_>` in the current scope
  --> $DIR/unresolved-iac-2.rs:10:25
   |
LL | struct Foo<'a> {
   | -------------- associated constant `X` not found for this struct
...
LL |     fn foo(_: [u8; Foo::X]) {}
   |                         ^ associated item not found in `Foo<'_>`
   |
   = note: the associated constant was found for
           

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0220`.
